### PR TITLE
docs: updated NX docs with azure react plugin docs

### PR DESCRIPTION
**Pull request by KarimHmaissi**
NX Azure React Plugin docs

### CODE REVIEW

File: docs/nx/azure-react/generator-app-insights-web.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+- Tracking of router changes
+- React components usage statistics
+
+It requires the connection string to be set in an env variable, from which it will be read.
+
+## Usage
+
+```bash
+nx generate @ensono-stacks/azure-react:app-insights-web --name=<app-name> --connectionString=<KEY_NAME> --directory=<directory where the lib is placed> --importPath=<import path for the library> --tags=<tags to the library>
```
Comment (Danphillipz) [inline docs/nx/azure-react/generator-app-insights-web.md:18]: Don't think we need to list all cli arguments here in addition to the table below. That's assuming the plugin prompts with questions to gather information for the required values. (Makes it nice and easy to just hit copy/paste then)
```suggestion
nx generate @ensono-stacks/azure-react:app-insights-web
```

File: docs/nx/azure-react/plugin-information.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - generator
+  - executor
+  - appinsights
+---
+
+import GeneratorAppInsights from './generator-app-insights-web.md';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Creates a ReactJS application and configures your ReactJS application to use various Azure services:
```
Comment (Danphillipz) [inline docs/nx/azure-react/plugin-information.md:21]: Does it create a ReactJS app? I thought it just enhances an existing one with our additions?

File: docs/nx/azure-react/generator-app-insights-web.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+<summary>Installs and configures a ReactJS app with App Insights.</summary>
+
+This generator will add and configure [applicationinsights reactjs](https://www.npmjs.com/package/@microsoft/applicationinsights-react-js) and [applicationinsights web](https://www.npmjs.com/package/@microsoft/applicationinsights-web) npm packages for you.
+
+This enables the following:
+
+- Tracking of router changes
+- React components usage statistics
+
+It requires the connection string to be set in an env variable, from which it will be read.
```
Comment (Danphillipz) [inline docs/nx/azure-react/generator-app-insights-web.md:13]: Could we move this into the usage sections and then have an admonition stating exactly what env variable needs to be set?
E,g

:::caution

App insights requires the `X` environment variable to be set to......x?

:::

File: docs/nx/azure-react/generator-app-insights-web.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+- Installs and adds microsoft applicationinsights packages to package.json
+```json title="/package.json"
+  "dependencies": {
+    ...OtherDependencies
+    "@microsoft/applicationinsights-react-js": "3.4.0",
+    "@microsoft/applicationinsights-web": "2.8.9",
+  }
+}
+```
+
```
Comment (Danphillipz) [inline docs/nx/azure-react/generator-app-insights-web.md:100]: Is this the last step for the user? Do they not need to do any additional configuration? Do they need to import it to an app somewhere or is everything automated?

**Review by Danphillipz — CHANGES_REQUESTED** [review]
Could we also get this added to the docs/nx/nx_stacks#stacks-plugins table please?

File: docs/nx/azure-react/plugin-information.md (hunk 2 above)
Comment (KarimHmaissi) [reply docs/nx/azure-react/plugin-information.md:21]: It does it runs the NX react generator

`import { libraryGenerator } from '@nrwl/react';`

`  // Use the existing library generator
    await libraryGenerator(tree, {
        ...options,
        linter: Linter.EsLint,
        style: 'none',
    });`

Comment (KarimHmaissi) [reply docs/nx/azure-react/plugin-information.md]: Ah I misunderstood the code. It actually does neither. It creates a ReactJS library which can then be imported into an existing react app. Updated the docs to reflect this

File: docs/nx/azure-react/generator-app-insights-web.md (hunk 4 above)
Comment (KarimHmaissi) [reply docs/nx/azure-react/generator-app-insights-web.md:106]: They will need to import the provider and wrap their app with it. Added some additional docs to explain this and a link to the official MS docs for more info

File: docs/nx/azure-react/generator-app-insights-web.md (hunk 3 above)
Comment (KarimHmaissi) [reply docs/nx/azure-react/generator-app-insights-web.md]: Added the admonition to explain how to set the env variable and a link to the MS docs regarding connection strings

**Review by Danphillipz — APPROVED** [review]

**Pull request merged**